File: src/main/java/me/lokka30/levelledmobs/misc/ChunkKillInfo.java
package me.lokka30.levelledmobs.misc;

import java.time.Instant;
import java.util.HashMap;
import java.util.Map;
import java.util.Set;
import org.jetbrains.annotations.NotNull;

/**
 * Records entity deaths for use in the chunk kill max feature
 *
 * @author stumper66
 * @since 3.4.0
 */
public class ChunkKillInfo {

    public ChunkKillInfo() {
        this.entityCounts = new HashMap<>();
    }

    // timestamp of death, max cooldown time
    final public @NotNull Map<Instant, Integer> entityCounts;

    public @NotNull Set<Map.Entry<Instant, Integer>> getEntrySet() {
        return entityCounts.entrySet();
    }

    public boolean isEmpty() {
        return this.entityCounts.isEmpty();
    }

    public int getCount() {
        return this.entityCounts.size();
    }

    public String toString() {
        return entityCounts.toString();
    }
}

File: levelledmobs-plugin/src/main/kotlin/io/github/arcaneplugins/levelledmobs/result/ChunkKillInfo.kt
package io.github.arcaneplugins.levelledmobs.result

import java.time.Instant

/**
 * Records entity deaths for use in the chunk kill max feature
 *
 * @author stumper66
 * @since 3.4.0
 */
class ChunkKillInfo {
    // timestamp of death, max cooldown time
    val entityCounts = mutableMapOf<Instant, Int>()

    val entrySet: Set<Map.Entry<Instant, Int>>
        get() = entityCounts.entries

    val isEmpty: Boolean
        get() = entityCounts.isEmpty()

    val count: Int
        get() = entityCounts.size

    override fun toString(): String {
        return entityCounts.toString()
    }
}